new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/LoggingUtil.java

```diff
@@ -55,75 +83,142 @@ public static void collectLogs(Object itInstance, List namespaces) {
    *
    * @param namespace in which to query cluster for artifacts
    * @param resultDir existing directory to write log files
-   * @throws IOException when writing to log files fail
-   * @throws ApiException when Kubernetes cluster query fails
    */
-  public static void generateLog(String namespace, Path resultDir) throws IOException, ApiException {
+  public static void collectLogs(String namespace, String resultDir) {
     logger.info("Collecting logs in namespace : {0}", namespace);
 
     // get service accounts
-    writeToFile(Kubernetes.listServiceAccounts(namespace), resultDir.toString(), namespace + "_sa.log");
+    try {
+      writeToFile(Kubernetes.listServiceAccounts(namespace), resultDir, namespace + "_sa.log");
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+    }
 
     // get namespaces
-    for (var ns: Kubernetes.listNamespacesAsObjects().getItems()) {
-      if (namespace.equals(ns.getMetadata().getName())) {
-        writeToFile(ns, resultDir.toString(), namespace + "_ns.log");
+    try {
+      for (var ns : Kubernetes.listNamespacesAsObjects().getItems()) {
+        if (namespace.equals(ns.getMetadata().getName())) {
+          writeToFile(ns, resultDir, namespace + "_ns.log");
+        }
       }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
     }
 
     // get pvc
-    writeToFile(Kubernetes.listPersistentVolumeClaims(namespace), resultDir.toString(), namespace + "_pvc.log");
-
-    // get pv based on the weblogic.domainUID in pvc
-    for (var pvc : Kubernetes.listPersistentVolumeClaims(namespace).getItems()) {
-      if (pvc.getMetadata() != null
-          && pvc.getMetadata().getLabels() != null
-          && pvc.getMetadata().getLabels().get("weblogic.domainUID") != null) {
-        String label = pvc.getMetadata().getLabels().get("weblogic.domainUID");
-        writeToFile(Kubernetes.listPersistentVolumes(
-            String.format("weblogic.domainUID in (%s)", label)), resultDir.toString(), label + "_pv.log");
+    try {
+      writeToFile(Kubernetes.listPersistentVolumeClaims(namespace), resultDir, namespace + "_pvc.log");
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+    }
+
+    // get pv configuration and pv files based on the weblogic.domainUid label in pvc
+    try {
+      for (var pvc : Kubernetes.listPersistentVolumeClaims(namespace).getItems()) {
+        String label = Optional.ofNullable(pvc)
```

Comment:
not sure if this is reliable 
spec.volumeName and spec.storageClassName might be better 
and/or get the pv and check its claim name matches

Reply:
changed the implementation to match storageclass, volume name and claim name to find the PV used by the test in that namespace.